src/idpi/config.py

```diff
@@ -0,0 +1,64 @@
+"""global configuration for idpi."""
+
+# Standard library
+from typing import Any, Literal
+
+config: dict = {}
+
+
+class set:
+    """Temporarily set configuration values within a context manager.
+
+    Parameters
+    ----------
+    config : dict, optional
+        dictionary object use to hold the configuration.
+        Default will use the config object in this module
+    **kwargs :
+        the configuration key-value pairs to set.
+
+    """
+
+    def __init__(self, config: dict = config, **kwargs):
```

Comment:
Relying on the default capture of a mutable variable to share state is a bit an unusual practice.